core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2079,7 +2079,7 @@ package object config {
         "conjunction with the server side flag spark.shuffle.server.mergedShuffleFileManagerImpl " +
         "which needs to be set with the appropriate " +
         "org.apache.spark.network.shuffle.MergedShuffleFileManager implementation for push-based " +
-        "shuffle to be enabled")
+        "shuffle to be enabled. Push-based shuffle is not yet supported.")
       .version("3.1.0")
```

Comment:
This configuration is added at 3.1.0 and do we want to add comments like `Push-based shuffle is not yet supported` at 3.1.3?

Reply:
In 3.1, multiple critical pieces of push-based shuffle were not there, for example SPARK-32922, SPARK-32921, and SPARK-32920. So, even if anyone even tried to turn it on, it would just not work. 
However, in 3.2 we are just missing 2 correctness fixes which involve modifying push-based shuffle protocols: #33078 and #33034. Without these 2 push-based shuffle still works but this is what we want to prevent.

Given this I don't think it makes a difference if we add these comments to branches of version < 3.2. WDYT?